File: financial-domain-starter/src/main/java/com/nixsolutions/financial/security/verifier/SecretAwareJwtVerifier.java
package com.nixsolutions.financial.security.verifier;

import static org.apache.commons.lang3.StringUtils.isEmpty;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;
import com.nixsolutions.financial.security.SecurityProperties;
import com.nixsolutions.financial.security.exception.InvalidTokenException;
import io.jsonwebtoken.Claims;
import io.jsonwebtoken.Jwts;

@Component
@ConditionalOnProperty(prefix = "financial-domain.security", name = "jwtSecret")
public class SecretAwareJwtVerifier implements JwtVerifier
{
  private SecurityProperties securityProperties;

  public SecretAwareJwtVerifier(SecurityProperties securityProperties)
  {
    this.securityProperties = securityProperties;
  }

  @Override
  public Claims parseToken(String token) throws InvalidTokenException
  {
    if (isEmpty(token))
    {
      throw new InvalidTokenException();
    }

    try
    {
      return Jwts.parser()
          .setSigningKey(securityProperties.getJwtSecret().getBytes())
          .parseClaimsJws(token)
          .getBody();
    }
    catch (Exception ex)
    {
      throw new InvalidTokenException(ex);
    }
  }
}

File: financial-domain-starter/src/main/kotlin/com/nixsolutions/financial/security/verifier/SecretAwareJwtVerifier.kt
package com.nixsolutions.financial.security.verifier

import com.nixsolutions.financial.security.SecurityProperties
import com.nixsolutions.financial.security.exception.InvalidTokenException
import io.jsonwebtoken.Claims
import io.jsonwebtoken.Jwts
import org.apache.commons.lang3.StringUtils.isEmpty
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.stereotype.Component

@Component
@ConditionalOnBean(SecurityProperties::class)
class SecretAwareJwtVerifier(private val securityProperties: SecurityProperties) : JwtVerifier {

  @Throws(InvalidTokenException::class)
  override fun parseToken(token: String?): Claims {
    if (isEmpty(token)) {
      throw InvalidTokenException("Token is empty!")
    }

    try {
      return Jwts.parser()
          .setSigningKey(securityProperties.jwtSecret?.toByteArray())
          .parseClaimsJws(token)
          .body
    } catch (ex: Exception) {
      throw InvalidTokenException(ex.message ?: "Wrong token", ex)
    }
  }
}
